picard/ui/edittagdialog.py

```diff
@@ -514,30 +514,18 @@ def _modified_tag(self):
             list(self.metadata_box.tag_diff.new[self.tag]) or [""]
         )
 
-    def _cleanup_modified_tags(self):
-        """Remove empty values from modified tags."""
+    def _modified_tags(self):
+        """Generate each modified tag and its non-empty values."""
         for tag, values in self.modified_tags.items():
-            self.modified_tags[tag] = [v for v in values if v]
-
-    def _update_object_metadata(self, obj, modified_tags):
-        """Update the metadata for a single object with the modified tags.
-
-        Args:
-            obj: The object to update
-            modified_tags: Iterable of (tag, values) pairs to update
-        """
-        for tag, values in modified_tags:
-            obj.metadata[tag] = list(values)
-        obj.update()
+            yield (tag, list(v for v in values if v))
```

Comment:
If it's supposed to be a map I'd use the list comprehension directly, instead of iterator and `list()`:

```
yield (tag, [v for v in values if v])
```

Reply:
Done, rebased